README.md

```diff
@@ -1,11 +1,14 @@
-# Rancher Image Mirror
+# Rancher Artifact Mirror
 
-This repo is dedicated to mirroring images from other organizations to the
+This repo is dedicated to mirroring artifacts from other organizations to the
 places they need to be for Rancher and its associated projects to use them.
 
+> [!NOTE]
+> This repository used to be called `image-mirror`; It was renamed to better reflect its usecases, see [this](https://github.com/rancher/rancher/issues/52259) issue for more information
```

Comment:
nit: spelling/grammar:
- "usecases" -> "use cases"
- "`image-mirror`; It" -> "`image-mirror`; it"
- Could use a period on the end of the sentence